usersession/agent/rest_api_test.go

```diff
@@ -903,3 +907,150 @@ func (s *restSuite) TestPostCloseRefreshNotification(c *C) {
+	c.Check(icon, Equals, "")
+}
+
+func (s *restSuite) TestGuessAppIconPrefixDifferentApp(c *C) {
+	si := createSnapInfo("snap1")
+	addAppToSnap(c, si, "app1", false, "iconname")
+	icon := agent.GuessAppIcon(si)
+	c.Check(icon, Equals, "iconname")
+}
+
+func (s *restSuite) TestGuessAppIconPrefixEqualApp(c *C) {
+	si := createSnapInfo("app1")
+	addAppToSnap(c, si, "app1", false, "iconname1")
+	addAppToSnap(c, si, "app2", false, "iconname2")
+	icon := agent.GuessAppIcon(si)
+	c.Check(icon, Equals, "iconname1")
+}
+
+func (s *restSuite) TestGuessAppIconServicePrefixEqualApp(c *C) {
+	si := createSnapInfo("app1")
+	addAppToSnap(c, si, "app1", true, "iconname")
+	icon := agent.GuessAppIcon(si)
+	c.Check(icon, Equals, "")
+}
+
+func (s *restSuite) TestGuessAppIconServicePrefixDifferentApp(c *C) {
+	si := createSnapInfo("snap1")
+	addAppToSnap(c, si, "app1", true, "iconname")
+	icon := agent.GuessAppIcon(si)
+	c.Check(icon, Equals, "")
+}
+
+func (s *restSuite) TestGuessAppIconServiceTwoApps(c *C) {
+	si := createSnapInfo("app1")
+	addAppToSnap(c, si, "app1", true, "iconname1")
+	addAppToSnap(c, si, "app2", false, "iconname2")
+	icon := agent.GuessAppIcon(si)
+	c.Check(icon, Equals, "iconname2")
+}
+
+func (s *restSuite) TestGuessAppIconServiceTwoAppsServices(c *C) {
+	si := createSnapInfo("app1")
+	addAppToSnap(c, si, "app1", true, "iconname1")
+	addAppToSnap(c, si, "app2", true, "iconname2")
+	icon := agent.GuessAppIcon(si)
+	c.Check(icon, Equals, "")
+}
+
+func (s *restSuite) TestGuessAppIconServiceTwoAppsOneServicePrefixDifferent(c *C) {
+	si := createSnapInfo("snap1")
+	addAppToSnap(c, si, "app1", true, "iconname1")
+	addAppToSnap(c, si, "app2", false, "iconname2")
+	icon := agent.GuessAppIcon(si)
+	c.Check(icon, Equals, "iconname2")
+}
+
+func (s *restSuite) TestGuessAppIconTwoAppsPrefixDifferent(c *C) {
+	si := createSnapInfo("snap1")
+	addAppToSnap(c, si, "app1", false, "iconname1")
+	addAppToSnap(c, si, "app2", false, "iconname2")
+	icon := agent.GuessAppIcon(si)
+	if (icon != "iconname1") && (icon != "iconname2") {
+		c.Fail()
+	}
+}
+
+func (s *restSuite) TestPostCloseRefreshNotificationWithIconDefault(c *C) {
+	dirs.SetRootDir(c.MkDir())
+
+	snap.MockSanitizePlugsSlots(func(snapInfo *snap.Info) {})
+	// add a notification first
+	mockYaml := `
+name: snap-name
+apps:
+  other-app:
+    command: /bin/foo
+  snap-name:
+    command: /bin/foo
+`
+	snaptest.MockSnapCurrent(c, mockYaml[1:], &snap.SideInfo{
+		Revision: snap.R("42"),
+	})
+
+	desktopEntry := `
+[Desktop Entry]
+Icon=foo.png
+`
+	os.MkdirAll(dirs.SnapDesktopFilesDir, 0755)
+	c.Assert(os.WriteFile(filepath.Join(dirs.SnapDesktopFilesDir, "snap-name_snap-name.desktop"), []byte(desktopEntry[1:]), 0644), IsNil)
+	refreshInfo := &client.FinishedSnapRefreshInfo{InstanceName: "snap-name"}
+	s.testPostFinishRefreshNotificationBody(c, refreshInfo)
+
+	notifications := s.notify.GetAll()
+	c.Assert(notifications, HasLen, 1)
+	n := notifications[0]
+	// boring stuff is checked above
```

Comment:
tiny nitpick, I think this comment makes more sense above the Icon check below, no?

Reply:
Done.